src/cargo/core/resolver/encode.rs

```diff
@@ -199,34 +204,49 @@ fn build_path_deps(ws: &Workspace) -> HashMap<String, SourceId> {
         visited.insert(member.package_id().source_id().clone());
     }
     for member in members.iter() {
-        build(member, ws.config(), &mut ret, &mut visited);
+        build_pkg(member, ws.config(), &mut ret, &mut visited);
+    }
+    for (_, deps) in ws.root_patch() {
+        for dep in deps {
+            build_dep(dep, ws.config(), &mut ret, &mut visited);
+        }
+    }
+    for &(_, ref dep) in ws.root_replace() {
```

Comment:
Hm, previously this looked at replace sections of other packages, so this might be a change of behavior. Is this intentional? I thinks it might cause problems for someone...

Perhaps we should leave replace behavior as is, add a test for it, and change only patch?

Reply:
Yeah it's intentional to avoid looking at replace/patch in all packages. That was actually a bug that was never intended! I don't think this'll cause problems because nothing else in Cargo looks at those sections, so it won't cause spurious updates or anything like that